batchservice/src/main/java/nu/yona/server/batch/quartz/JobController.java

```diff
@@ -0,0 +1,148 @@
+
+	@RequestMapping(value = "/", method = RequestMethod.GET)
+	@ResponseBody
+	public HttpEntity<Resources<JobResource>> getAllJobs()
+	{
+		return new ResponseEntity<>(
+				createJobsCollectionResource(jobManagementService.getAllJobs(), getAllJobsLinkBuilder().withSelfRel()),
+				HttpStatus.OK);
+	}
+
+	@RequestMapping(value = "/{group}/", method = RequestMethod.GET)
+	@ResponseBody
+	public HttpEntity<Resources<JobResource>> getJobsInGroup(@PathVariable String group)
+	{
+		return new ResponseEntity<>(createJobsCollectionResource(jobManagementService.getJobsInGroup(group),
+				getJobsInGroupLinkBuilder(group).withSelfRel()), HttpStatus.OK);
+	}
+
+	@RequestMapping(value = "/{group}/", method = RequestMethod.PUT)
+	@ResponseBody
+	public HttpEntity<Resources<JobResource>> updateJobGroup(@PathVariable String group, @RequestBody Set<JobDto> jobs)
+	{
+		return new ResponseEntity<>(createJobsCollectionResource(jobManagementService.updateJobGroup(group, jobs),
+				getJobsInGroupLinkBuilder(group).withSelfRel()), HttpStatus.OK);
+	}
+
+	@RequestMapping(value = "/{group}/{name}", method = RequestMethod.GET)
+	@ResponseBody
+	@ResponseStatus(HttpStatus.OK)
+	public JobResource getJob(@PathVariable String group, @PathVariable String name)
+	{
+		return new JobResourceAssembler().toResource(jobManagementService.getJob(name, group));
+	}
+
+	@RequestMapping(value = "/{group}/", method = RequestMethod.POST)
+	@ResponseBody
+	@ResponseStatus(HttpStatus.CREATED)
+	public JobResource addJob(@PathVariable String group, @RequestBody JobDto job)
+	{
+		job.setGroup(group);
+		return new JobResourceAssembler().toResource(jobManagementService.addJob(group, job));
+	}
+
+	private static Resources<JobResource> createJobsCollectionResource(Set<JobDto> allJobs, Link link)
+	{
+		return new Resources<>(new JobResourceAssembler().toResources(allJobs), link);
+	}
+
+	private static ControllerLinkBuilder getAllJobsLinkBuilder()
+	{
+		JobController methodOn = methodOn(JobController.class);
+		return linkTo(methodOn.getAllJobs());
+	}
+
+	private static ControllerLinkBuilder getJobsInGroupLinkBuilder(String group)
+	{
+		JobController methodOn = methodOn(JobController.class);
+		return linkTo(methodOn.getJobsInGroup(group));
+	}
+
+	static ControllerLinkBuilder getJobLinkBuilder(String group, String name)
+	{
+		JobController methodOn = methodOn(JobController.class);
+		return linkTo(methodOn.getJob(group, name));
+	}
+
+	static class JobResource extends Resource<JobDto>
+	{
+		public JobResource(JobDto job)
+		{
+			super(job);
+		}
+	}
+
+	public static class JobResourceAssembler extends ResourceAssemblerSupport<JobDto, JobResource>
+	{
+		public JobResourceAssembler()
+		{
+			super(JobController.class, JobResource.class);
+		}
+
+		@Override
+		public JobResource toResource(JobDto job)
+		{
+			JobResource jobResource = instantiateResource(job);
+			addSelfLink(jobResource);
+			addEditLink(jobResource);/* always editable */
```

Comment:
Fair point. I wanted to bring the editing at the job level, but then implemented it at the group level. I'm removing the ``addEditLink`` method.